src/charm.py

```diff
@@ -7,14 +7,29 @@
 import logging
 import secrets
 import string
+from time import sleep
 
+from lightkube import ApiError, Client, codecs
+from lightkube.resources.core_v1 import Pod
 from ops.charm import ActionEvent, CharmBase, WorkloadEvent
 from ops.main import main
-from ops.model import ActiveStatus, Relation, WaitingStatus
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    Relation,
+    WaitingStatus,
+)
 from ops.pebble import Layer
+from tenacity import RetryError
+
+from patroni import Patroni
 
 logger = logging.getLogger(__name__)
 
+PEER = "postgresql-replicas"
+STORAGE_PATH = "/var/lib/postgresql/data"
```

Comment:
@marceloneppel 
noob question: since the storage_path is being defined in the metadata, could it be retrieved in some different manner to avoid duplication?

Reply:
Good call - you might be able to use the `CharmMeta` class in `ops` for this